src/releaseNotesProvider.ts

```diff
@@ -0,0 +1,249 @@
+import { env, ExtensionContext } from "vscode";
+import * as vscode from "vscode";
+import * as semver from "semver";
+import { Remarkable } from "remarkable";
+import { fetchFrom } from "./util";
+import { Either, makeLeft, makeRight } from "./types";
+
+const versionKey = "metals-server-version";
+
+/**
+ * Show release notes if possible, swallow errors since its not a crucial feature.
+ * Treats snapshot versions like 0.11.6+67-926ec9a3-SNAPSHOT as a 0.11.6.
+ */
+export async function showReleaseNotesIfNeeded(
+  context: ExtensionContext,
+  serverVersion: string,
+  outputChannel: vscode.OutputChannel
+) {
+  try {
+    // context.globalState.update(versionKey, "0.11.5");
+    const result = await showReleaseNotes(context, serverVersion);
+    if (result.kind === "left") {
+      const msg = `Release notes was not shown: ${result.value}`;
+      outputChannel.appendLine(msg);
+    }
+  } catch (error) {
+    outputChannel.appendLine(
+      `Error, couldn't show release notes for Metals ${serverVersion}`
+    );
+    outputChannel.appendLine(`${error}`);
+  }
+}
+
+async function showReleaseNotes(
+  context: ExtensionContext,
+  currentVersion: string
+): Promise<Either<string, void>> {
+  const state = context.globalState;
+  state.update(versionKey, "0.11.5");
+
+  const remote = isRemote();
+  if (remote.kind === "left") {
+    return remote;
+  }
+
+  const version = getVersion();
+  if (version.kind === "left") {
+    return version;
+  }
+
+  const releaseNotesUrl = await getMarkdownLink(version.value);
+  if (releaseNotesUrl.kind === "left") {
+    return releaseNotesUrl;
+  }
+
+  // actual logic starts here
+  await showPanel(version.value, releaseNotesUrl.value);
+  return makeRight(undefined);
+
+  // below are helper functions
+
+  async function showPanel(version: string, releaseNotesUrl: string) {
+    const panel = vscode.window.createWebviewPanel(
+      `scalameta.metals.whatsNew`,
+      `Metals ${version} release notes`,
+      vscode.ViewColumn.One
+    );
+
+    const releaseNotes = await getReleaseNotesMarkdown(
+      releaseNotesUrl,
+      context,
+      (uri) => panel.webview.asWebviewUri(uri)
+    );
+
+    panel.webview.html = releaseNotes;
+    panel.reveal();
+
+    // Update current device's latest server version when there's no value or it was a older one.
+    // Then sync this value across other devices.
+    state.update(versionKey, version);
+    state.setKeysForSync([versionKey]);
+
+    context.subscriptions.push(panel);
+  }
+
+  /**
+   * Don't show panel for remote environment because it installs extension on every time.
+   * TODO: what about wsl?
+   */
+  function isRemote(): Either<string, void> {
+    // const isWsl = env.remoteName === "wsl";
+    return env.remoteName == null
+      ? makeRight(undefined)
+      : makeLeft(`is a remote environment ${env.remoteName}`);
+  }
```

Comment:
@Arthurm1 could you try this PR out on Windows & wsl? I'm not sure how wsl is being treated and I don't have windows myself

Reply:
I've run
```
yarn install
yarn build
```
and then installed `metals-1.17.2.vsix` but when I try to start Metals I get...

![image](https://user-images.githubusercontent.com/5612295/173187205-d9d2f06d-44dd-4040-b2e3-25676a373739.png)

What step have I missed out?  Why would `remarkable` be missing as `yarn build` works

Reply:
I've run this only in debug extension mode so I'm not sure if this error happens also on other systems.
Try to remove node_moduels and run yarn install again

Reply:
I can also test that out. You can usually test it out by running inside VS Code, but I wonder if the issue you are hitting on is soemthing to do with wrongly specifying the dependencies? @kpodsiad did you test it out this was on mac?

Reply:
In debug extension mode I can start Metals and get...
```
2022.06.11 13:27:59 INFO  Started: Metals version 0.11.5-SNAPSHOT in workspace '/root/metals' for client Visual Studio Code 1.68.0.
2022.06.11 13:28:03 INFO  time: initialize in 3.66s
Initializing Scala Debugger
Release notes was not shown: is a remote environment wsl
2022.06.11 13:28:03 INFO  Flyway Community Edition 8.5.8 by Redgate
```

Is that what's wanted?

Reply:
Ok, I found what was wrong. I accidentally put `remarkable` at `devDependencies` and they're are present only at dev development but they aren't included in bundle create via `yarn build`. Now it should work!

@Arthurm1 in which output tab did you have such logs? I couldn't find anything useful and without your's scrrenshot it would take much longer to find issue. Thanks for it!

@Arthurm1 I changed logic to show release notes on `wsl`. Could you try to see if it works and then, comment `state.update(versionKey, "0.11.5");` (line 39) and see if release notes are showed only once and subsequent windows don't display them.

Reply:
I can build and install it now.

No release notes are shown and I get this...
![image](https://user-images.githubusercontent.com/5612295/173243652-81db0b76-3bf3-49d4-ae98-4527ff2790a3.png)


Same thing after commenting out `state.update(versionKey, "0.11.5");`

This looks correct to me because the code is...
![image](https://user-images.githubusercontent.com/5612295/173243975-f4bb5145-bde6-4e10-ae34-58494ad496b2.png)
So it will never get far enough in `showReleaseNotes` to show the release notes

Is it supposed to show the release notes for WSL?

Reply:
> Is it supposed to show the release notes for WSL?

Yes, we should show them. 

I realized that condition in `isRemote` was broken, now it should work correctly.

Reply:
Now I get release notes and if I comment out line 39 then I get `Release notes was not shown: do not show release notes for an older version`

So looking good.